src/renderer/components/ModelSelect.tsx

```diff
@@ -88,78 +103,741 @@
-  dropdownClassName?: string;
-  tooltipId?: string;
-  showParentMenu?: React.Dispatch<React.SetStateAction<boolean>>;
+  model: Model;
+  computedModelDataMap: Map<string, ComputedModelData>;
+  sortBy: SortMethod;
+  isFeatherless: boolean;
+  showDescriptionOn: string | null;
+  setShowDescriptionOn: React.Dispatch<React.SetStateAction<string | null>>;
+  recomputeExpandedItemHeight: () => void;
 }) {
-  const dispatch = useAppDispatch();
-  const t = useAppSelector((state: RootState) => state.app.translations);
-  const [showModels, setShowModels] = useState(false);
+  const description = useMemo(() => {
+    return model.featherless?.readme ?? model.description ?? "";
+  }, [model.featherless?.readme, model.description]);
+  const showDescription = useMemo(() => {
+    return showDescriptionOn === model.id;
+  }, [showDescriptionOn, model.id]);
+  const convertMarkdownToComponent = useConvertMarkdownToComponent(vscode);
+  const backendMessenger = useMessenger(vscode);
   const settings = useAppSelector(
     (state: RootState) => state.app.extensionSettings
   );
-  const apiKeyStatus = useAppSelector(
-    (state: RootState) => state.app?.apiKeyStatus
+  const fetchModelDescription = useCallback(
+    async (model: Model) => {
+      if (model.featherless?.status) {
+        backendMessenger.sendGetModelDetails(model.id);
+      }
+    },
+    [model]
   );
-  const models: Model[] = useAppSelector(
-    (state: RootState) => state.app.models
+  const fetchModelDataDebounced = useDebounce(fetchModelDescription, 1000);
+  const descriptionIconClickHandler = useCallback(
+    (event: any, model: Model) => {
+      event.stopPropagation();
+
+      const show = showDescription ? null : model.id;
+
+      setShowDescriptionOn(show);
+
+      if (show) {
+        // Fetch readme on featherless models if it's not already fetched
+        if (model.featherless?.status && !model.featherless?.readme) {
+          // Use debounced function to avoid inadvertantly spamming the API
+          fetchModelDataDebounced(model);
+        }
+      } else {
+        // Unfocus the button when the description is hidden
+        event.currentTarget.blur();
+      }
+    },
+    [showDescription]
   );
-  const sync = useAppSelector((state: RootState) => state.app.sync);
-  const [filteredModels, setFilteredModels] = useState<Model[]>(models);
+  // recomputeExpandedItemHeight when the model's description changes
+  useEffect(() => {
+    const description = model.featherless?.readme ?? model.description;
+
+    if (showDescription && description && description.length > 0) {
+      recomputeExpandedItemHeight();
+    }
+  }, [model.featherless?.readme, model.description, showDescription]);
+  const modelDescriptionComponent = useMemo(() => {
+    if (showDescription) {
+      let description = model.featherless?.readme ?? model.description ?? "";
+
+      // Remove any HTML comments
+      description = description.replace(/<!--.*?-->/gs, "");
```

Comment:
Not a security issue—comments showing just didn't look good, but this fix is added anyway in case string replacement is later added that is security critical.